src/USER-MISC/pair_agni.cpp

```diff
@@ -347,132 +358,123 @@ double PairAGNI::init_one(int i, int j)
+      params[curparam].sigma = values.next_double();
+
+      values = reader.next_values(2);
+      values.next_string(); // ignore
+      values.next_double(); // ignore
+
+      values = reader.next_values(2);
+      values.next_string(); // ignore
+      params[curparam].b = values.next_double();
+
+      values = reader.next_values(1);
+      values.next_string(); // ignore
       wantdata = curparam;
       curparam = -1;
-    } else if ((curparam >=0) && (nwords == 2) && (strcmp(words[0],"Rc") == 0)) {
-      params[curparam].cut = atof(words[1]);
-    } else if ((curparam >=0) && (nwords == 2) && (strcmp(words[0],"Rs") == 0)) {
-      ; // ignored
-    } else if ((curparam >=0) && (nwords == 2) && (strcmp(words[0],"neighbors") == 0)) {
-      ; // ignored
-    } else if ((curparam >=0) && (nwords == 2) && (strcmp(words[0],"sigma") == 0)) {
-      params[curparam].sigma = atof(words[1]);
-    } else if ((curparam >=0) && (nwords == 2) && (strcmp(words[0],"lambda") == 0)) {
-      params[curparam].lambda = atof(words[1]);
-    } else if ((curparam >=0) && (nwords == 2) && (strcmp(words[0],"b") == 0)) {
-      params[curparam].b = atof(words[1]);
-    } else if ((curparam >=0) && (nwords == 2) && (strcmp(words[0],"n_train") == 0)) {
-      params[curparam].numtrain = atoi(words[1]);
-    } else if ((curparam >=0) && (nwords > 1) && (strcmp(words[0],"eta") == 0)) {
-      params[curparam].numeta = nwords-1;
-      params[curparam].eta = new double[nwords-1];
-      for (i = 0, j = 1 ; j < nwords; ++i, ++j)
-        params[curparam].eta[i] = atof(words[j]);
-    } else if (params && (wantdata >=0) && (nwords == params[wantdata].numeta+3)) {
-      n = (int) atof(words[0]);
-      for (i = 0; i < params[wantdata].numeta; ++i) {
-        params[wantdata].xU[i][n] = atof(words[i+1]);
-      }
-      params[wantdata].yU[n] = atof(words[params[wantdata].numeta+1]);
-      params[wantdata].alpha[n] = atof(words[params[wantdata].numeta+2]);
 
-    } else {
-      if (comm->me == 0)
-        error->warning(FLERR,"Ignoring unknown content in AGNI potential file.");
+      if (params && wantdata >=0){
+        for(j = 0; j < params[wantdata].numtrain; j++){
+          values = reader.next_values(params[wantdata].numeta + 2);
+          for (i = 0; i < params[wantdata].numeta; ++i) 
+            params[wantdata].xU[i][j] = values.next_double();
+          values.next_double(); // ignore
+          params[wantdata].alpha[j] = values.next_double();   
+        } 
+      }else
+        error->all(FLERR,"Invalid AGNI potential file");
+    } catch (TokenizerException &e) {
+      error->one(FLERR, e.what());
     }
   }
-
-  delete [] words;
+  
+  MPI_Bcast(&nparams, 1, MPI_INT, 0, world);
+  MPI_Bcast(&atomic_feature_version, 1, MPI_INT, 0, world);
+  if(comm->me != 0) {
+    params = memory->create(params,nparams,"pair:params");
+    memset(params,0,nparams*sizeof(Param));
+  }
+  MPI_Bcast(params, nparams*sizeof(Param), MPI_BYTE, 0, world);
 }
```

Comment:
A significant difference between the old style file reader and the new reader is that you now do reading *and* parsing only on MPI rank 0, whereas previously you would read a line then broadcast and then parse concurrently on all MPI ranks in parallel.